lib/msf/core/post/vcenter/vcenter.rb

```diff
@@ -0,0 +1,711 @@
+          return nil if out.nil?
+
+          out.split("\n")
+        end
+
+        #
+        # Returns a list of hashes for the vecs store
+        # @param vecs_store [String] the store to get entries from
+        # @return [Array] of hashes, nil on error
+        #
+        def get_vecs_entries(vecs_store)
+          return nil unless command_exists? vecs_bin
+
+          out = cmd_exec("#{vecs_bin} entry list --store #{vecs_store}")
+          return nil if out.nil?
+
+          # time to process this beast. its roughly " : " delimited, but things like certificates are multi-line
+          delimiter = " :\t"
+          output = []
+          current_entry = {}
+          last_key = ''
+          out = out.split("\n")
+          out.each do |line|
+            # handle anything that looks to be a continuation of the last key
+            unless line.include? delimiter
+              current_entry[last_key] = "#{current_entry[last_key]}\n#{line}"
+              next
+            end
+
+            line = line.split(delimiter).map(&:strip)
+            key = line.shift
+            value = line.join(delimiter)
+            last_key = key
+            next if key.include? 'Number of entries in store' # heading for output
+
+            # Alias is assumed first line of an entry, so append any non-blank previous entries to our output
+            if key == 'Alias' && !current_entry.empty?
+              output.append(current_entry)
+              current_entry = { key => value }
+              next
+            end
+            current_entry[key] = value
+          end
+          output.append(current_entry) unless current_entry.empty?
+          return output
+
+          nil
+        end
+
+        #
+        # Returns a private key for an alias in a vecs store
+        # @param vecs_store [String] the store to get entries from
+        # @return [OpenSSL::PKey::RSA] of content, nil on error
+        #
+        def get_vecs_private_key(vecs_store, entry_alias)
+          return nil unless command_exists? vecs_bin
+
+          key_b64 = cmd_exec("#{vecs_bin} entry getkey --store #{vecs_store} --alias #{entry_alias}")
+          begin
+            return OpenSSL::PKey::RSA.new(key_b64)
+          rescue OpenSSL::PKey::PKeyError
+            nil
+          end
+          nil
+        end
+
+        #
+        # A helper function to return the command line statement string to connect to the postgress server
+        # @param pg_password [String] postgress password
+        # @param vcdb_user [String] virtual center database username
+        # @param vcdb_name [String] virtual center database name
+        # @param vcdb_host [String] virtual center hostname. Defaults to 'localhost'
+        # @return [String] a string to run on command line
+        #
+        def postgress_connect(pg_password, vcdb_user, vcdb_name, vcdb_host = 'localhost')
+          "PGPASSWORD=#{pg_password} #{psql_bin} -h '#{vcdb_host}' -U '#{vcdb_user}' -d '#{vcdb_name}'"
```

Comment:
pg_password should also be quoted here lest it contain space or double quote. It'll still break if there's a single quote but so will everything else.


```suggestion
          "PGPASSWORD='#{pg_password}' #{psql_bin} -h '#{vcdb_host}' -U '#{vcdb_user}' -d '#{vcdb_name}'"
```

Reply:
in theory this came in quoted from the module, but being its a lib there is no guarantee obviously.  I've taken out the quotes, and used an if statement to wrap if needed since the only module that calls this currently does the wrapping itself.